### DISCUSSION TITLE
Github only recognizes contributions I push from main branch of my project

### DISCUSSION THREAD

**Discussion by Herrefer** [community/community#123132] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I just picked up an old project to make improvements, but I realized that when I push my changes from new branches they don't count in the GitHub contributions table :( (except main/master push).
However, all new branches, changes, and commits DO appear in my repository.

**Comment by Herrefer** [comment, marked as answer]
it was a fool question, i resolved it 👍

**Reply by Gabriel-Matias07** [reply]
Como resolveu?

**Reply by Herrefer** [reply]
I think its because commits and changes only show in chart when you do it in default branch (usually main branch). So if you make changes in other branches it count only when you merge that branch with main.

**Reply by Gabriel-Matias07** [reply]
Right, thanks

**Reply by ab12gu** [reply]
@Herrefer is there any reason/benefit for this?

**Reply by Deepak-png981** [reply]
yup, I am curious too. What's the rational behind limiting it to Main or Master?

**Reply by Eduardo79Silva** [reply]
I believe (not too sure about this) that it is because otherwise it is not sure to be a "Contribution", the idea is not to track commits you made in any project but rather how many contributions you made that actually ended up in the project, therefore, in the default branch.
